### ISSUE TITLE
Notification drawer demo

### ISSUE THREAD

**Issue by cliffpyles** [patternfly/patternfly#35]
The Notification Drawer is a content delivery system such as events, tasks, and alerts. It is a self contained system that is viewable without having to navigate to another area of the application. Please note, there is currently a bug in the pf3 notification drawer with the dropdown menu ([see issue here](https://github.com/patternfly/patternfly/issues/1039)). We want to ensure to address this issue in this new implementation. 

Design folder: https://drive.google.com/open?id=1ZLl8gXAP7n5gkCdS6MN3W_Cf4lC1Ob-j

Updated wireframes: https://docs.google.com/document/d/1UH6J2Mf1Nqx4zUx-TzO6MCgPFq6NJatRb0kagmGN5yE/edit?usp=sharing

Component lifecycle: 
- [x] Interaction design requirements 
- [x] Visuals 
- [x] First checkpoint complete and approved 
- [ ] CSS 
- [ ] React
- [ ] Second checkpoint complete and approved 

Contacts:

**Comment by cliffpyles** [comment]
https://github.com/patternfly/patternfly-next/pull/36

**Comment by mceledonia** [comment]
[Working file](https://drive.google.com/open?id=1X4uepwcNhUiSyCTvProAMo_M9v-mWzAc)
[Static Screens](https://github.com/patternfly/patternfly-next/files/1834447/Notif_drawer_static.zip)

**Comment by cliffpyles** [comment]
Thanks @mceledonia! These look great!

**Comment by mceledonia** [comment]
Notification drawer sketch + mockup example
[Notification Drawer.zip](https://github.com/patternfly/patternfly-next/files/2002349/Notification.Drawer.zip)

**Comment by mcarrano** [comment]
@mceledonia Looks good, but just a few comments/questions:

* I'm OK with moving the "Mark All Read" and "Clear All" actions from the footer of the expanded section to the header. However the "Delete All" should be changed back to "Clear All" as the action is only to clear these items from the drawer, not to delete them permanently.

* I do have concern with using the bright color bands at the left of each notification to indicate severity. While I like this for an alert, it seems like an overwhelming amount of color when the drawer is open, but that's just my opinion. Would be interested in whether others had the same reaction.

* Regarding the link in the lower right hand corner, seems unlikely that we would have both the kabob and an action link, or if we did, they should be associated by proximity since the kabob would essentially be a place to hold more links/actions than could be exposed at once. 

* Curious why only one event includes a date/time stamp. Wouldn't each item have a title, description, and timestamp by default?

Let me know your thoughts.

**Comment by mceledonia** [comment]
@mcarrano Thanks for the feedback!

- I wasn't certain about the "clear all" action, so I am more than happy to change that back. I initially was considering a confirmation dialogue (are you sure?) but knowing that it only clears and doesn't actually delete, I don't think that's necessary either.
- The bright color bands were a choice to align with the alerts, and also to give some more visual prominence to the expanded area. The panel lacked a bit of hierarchy and felt a bit flat, check out the sketch file to see some alternate designs that had that problem. I agree it's a bit much seeing them together and I'm open to discussion around alternatives if we can come up with something better!
- This link/button text was moreso included because I saw it in most of the case studies I was looking at for inspiration. The idea here was that each notification entry would have a collapsed and expanded state, and this expanded state revealed a sort of primary action on the notification itself. I wasn't sure what kind of actions or links typically are in the kebab menu, if it's typically just one or two items we might consider bringing them out with the button text instead. However I was assuming it was multiple, in that case the thought was to bring the primary action to the forefront with that button text and leave the leftovers in the kebab menu.
- See above regarding expanded/collapsed state. I don't think including the time/date in the collapsed state is a bad idea! More of an oversight than a conscious decision to remove.

**Comment by mcarrano** [comment]
@mceledonia Good questions about the need for links vs. kabob actions. I will try to collect some examples from product to see how notification drawers are actually being used. I am curious what actions one can take here other than seeing full details of an alert. I think that having an expanded and collapsed state for a single alert is a good idea, however from looking at the mockup, it's not clear what the interaction is.

Let's continue to work on these details.

**Comment by kybaker** [comment]
@matthewcarleton 

[Here is the latest updates](https://redhat.invisionapp.com/share/HVJBE71B2DJ)
[Editable File](https://drive.google.com/open?id=1Hmt3IFCktv2fHj2E3F28Xm8VTNv40oeK)

**Comment by kybaker** [comment]
@matthewcarleton 

[Latest Sketch file with the updates](https://drive.google.com/open?id=1c86ICojc95DXTbtcX6ZvG5CEjx-jNPX7)

**Comment by mcarrano** [comment]
Here are some wireframes for the notification drawer based on requirements gathered from product teams: https://drive.google.com/open?id=1S5uwkqLSVbYvckopRYkWEIpXjZJEKb06  Also reference https://github.com/patternfly/patternfly-next/issues/299 for more details about intended interaction design.

**Comment by mcarrano** [comment]
Note that complex variation (with categories) is a candidate for deprecation. Simple notification drawer will satisfy most use cases.

**Comment by lwrigh** [comment]
In PF 4 is there any talk of notification settings?

**Comment by rachael-phillips** [comment]
Hi Laura, could you tell me more about your needs for notification settings and what you are looking for? That will help me answer your question.

**Comment by lwrigh** [comment]
Hi @rachael-phillips, in RHV we're trying to figure out a way of limiting the amount toast notifications that a user receives since sometimes they can sometimes be bombarded with 10+ toast notifications at a time and it can cover important action buttons or information. A couple of users have raised this concern as well. We've been exploring design ideas around adding a 'Do Not Disturb' function to the notifications but I feel like options like that are just short term fixes. I think to really try to solve the problem of the user receiving too many toast notifications at one time they need to have more of an ability to edit the settings of the sort of notifications they receive. The notification categories would definitely vary by product but I think it's an important thing to consider for trying to more effectively manage a user's notification settings.

**Comment by mcarrano** [comment]
So are you thinking we would add some type of a settings menu within the notification drawer @lwrigh ? I think that is doable. We have not implemented this component yet.

Also, we've talked about the need to aggregate notifications when they exceed a certain number. That's something I was working on from a design perspective.

**Comment by lwrigh** [comment]
@mcarrano Yeah I think adding a settings menu to the notification drawer would be a good addition. I think the aggregate notification idea is a good one as well. It definitely helps to make the volume of toasts more bearable.

**Comment by mcarrano** [comment]
I have created a set of updated wireframes that can be found here: https://drive.google.com/open?id=1nU9dpAbGiKSRIhK3GPx52xJQH_578YAB  These should be used as a basis for updating Visual designs. @rachael-phillips @kybaker FYI

**Comment by maryshak1996** [comment]
Based on @mcarrano 's updated wires and some communication with @kybaker , I updated visuals for this! I have mockups default and on-scroll states for both the 'simple' and 'complex' notification drawer. Those and their respective specs can be found in my prototype here: https://redhat.invisionapp.com/share/7FQDO4RP462#/345357487_Examples
Here's the gist of the simple and complex drawers:
Simple
![screen shot 2019-02-06 at 3 07 51 pm](https://user-images.githubusercontent.com/19824574/52370487-08b6dd00-2a21-11e9-8a32-e1f3c5b873c2.png)
Complex
![screen shot 2019-02-06 at 3 08 06 pm](https://user-images.githubusercontent.com/19824574/52370496-0f455480-2a21-11e9-8ac3-be189e97a906.png)

**Comment by mcarrano** [comment]
@maryshak1996 This is looking good. I left a few comments on the InVision doc.

**Comment by maryshak1996** [comment]
thanks @mcarrano ! I replied to your comments -- thanks for the feedback so far!

**Comment by maryshak1996** [comment]
@mcarrano @matthewcarleton I have updated my invision prototype with some updates, based on some great feedback from you guys and others :) 
Simple
<img width="540" alt="screen shot 2019-02-08 at 11 41 17 am" src="https://user-images.githubusercontent.com/19824574/52492075-7b959480-2b96-11e9-82c8-0957dae1385b.png">
Complex
<img width="538" alt="screen shot 2019-02-08 at 11 41 09 am" src="https://user-images.githubusercontent.com/19824574/52492086-80f2df00-2b96-11e9-8fe0-7f5e0257b044.png">
And I do want to also call some attention to how we are proposing indicating an unread notification in the masthead (this is also shown within the invision's "Visual Spec' page)
<img width="1077" alt="screen shot 2019-02-08 at 11 35 42 am" src="https://user-images.githubusercontent.com/19824574/52492166-a849ac00-2b96-11e9-98c6-197a1b217675.png">

**Comment by mcarrano** [comment]
This is looking great @maryshak1996 !  Just one question about icons. In the example I see the you changed the icon for notification settings to the sliders, but in the masthead we show a gear icon. It raises two questions: 1) Which is the proper icon for settings and should we use it consistently? 2) It does seem weird to see these two settings icons in proximity. The purpose of the settings icon in the header of the drawer was for notification settings specifically. But does it make sense for all settings to be accessed globally from the masthead? Or should these be differentiated? Would be interested to hear what others think.

Also, I think we want to start removing real people's names from these mockups because we will eventually publish these to the website. @gdoyle1 did we ever decide what to do about that in our mocks?

**Comment by gdoyle1** [comment]
@mcarrano I never ended up updating those in the master sketch file but I can do that. When we finish the symbol library, we will definitely use fake names!

**Comment by maryshak1996** [comment]
@mcarrano I was looking through what we had in the master sketch file icons page, and that page is calling for 'Settings' to use that icon. I'm not sure if that is just out of date, or if there is a greater consistency issue here.
@kybaker do you have any thoughts on that icon topic?

**Comment by maryshak1996** [comment]
@mcarrano @kybaker @gdoyle1 I did a little bit more iterating and I think that I can put the done stamp on the visual side of this! The invision prototype is up to date and here is my latest updated designs to take a look at: 
<img width="564" alt="screen shot 2019-02-12 at 2 14 43 pm" src="https://user-images.githubusercontent.com/19824574/52661704-9cc8ee80-2ed0-11e9-8307-d8a712602cf2.png">
<img width="956" alt="screen shot 2019-02-12 at 2 14 57 pm" src="https://user-images.githubusercontent.com/19824574/52661705-9e92b200-2ed0-11e9-8926-1b44da7f61c5.png">
Visual spec can be found on the invision link or in this PDF
[notification-drawer_visual-spec.pdf](https://github.com/patternfly/patternfly-next/files/2857296/notification-drawer_visual-spec.pdf)

**Comment by maryshak1996** [comment]
https://redhat.invisionapp.com/share/7FQDO4RP462

**Comment by gdoyle1** [comment]
@maryshak1996 I like the additional background color change for unreads! Looks great

**Comment by mcarrano** [comment]
Overall this looks great. I like the changes you made @maryshak1996 . I just had one question about the location of the footer in the complex drawer that I left in InVision.

**Comment by maryshak1996** [comment]
Thanks @gdoyle1 ! and @mcarrano thanks for pointing that out, what you suggested is what I had in mind in the first place and it was just an oversight on my part :) 
https://redhat.invisionapp.com/share/7FQDO4RP462
@mcoker I think that the design on this is complete, and here is the most up-to-date spec
[notification-drawer_visual-spec.pdf](https://github.com/patternfly/patternfly-next/files/2860841/notification-drawer_visual-spec.pdf)

**Comment by mcarrano** [comment]
Looks good to me @maryshak1996 . @mcoker can you take a look and let us know if you see any issues that would affect implementation of this? @rachael-phillips pending that, I think this can move to CSS as soon as someone can take it.

**Comment by rachael-phillips** [comment]
Thanks for letting me know! I'll be on the lookout for what @mcoker has to say!

**Comment by mcoker** [comment]
Thanks @maryshak1996, this looks great! A couple of questions:

* It looks like the individual notifications are what scrolls - whether there are just notifications in the drawer, or those notifications are in a notification group. What happens if the number of groups overflows the height of the drawer? Would that add an additional scrollbar, so you have an outer scroll to scroll the groups and an inner scroll to scroll notifications within a group? FWIW, looking at PF3, if the notification "tabs" (groups) overflow the window height, they're just hidden and you can't access them. (The scrollbar on the right in the screenshot is just for the overall page, not for the notification drawer)

<img width="454" alt="screen shot 2019-02-13 at 3 56 52 pm" src="https://user-images.githubusercontent.com/35148959/52746813-0b2cb000-2fa8-11e9-93af-d61fb9eaac62.png">

* If you click the kebab icon and open the dropdown of an alert in the drawer, is the dropdown menu expected to display inside of the drawer (if it's not in a group) and/or group (if it's in a group)? If the dropdown is expected to display outside of either, that might complicate things as it's difficult to get children of a scrollable area to display outside of the scrollable area, though I _think_ that would be possible to do with the react component (I would want to double check to make sure). Looks like PF3 displays the dropdown within the drawer/group as I would expect, which is easy to do.

<img width="449" alt="screen shot 2019-02-13 at 3 58 06 pm" src="https://user-images.githubusercontent.com/35148959/52747078-a0c83f80-2fa8-11e9-81ab-5870c642709c.png">

**Comment by mcarrano** [comment]
Let me try to answer these questions @mcoker . 

>     * What happens if the number of groups overflows the height of the drawer?
The likelihood of this is small. If groups are used at all (in most cases they won't be) it would normally be a small number (less than 5). In most cases this will work with the current accordion approach. I would not be in favor of extra work to add an additional scrollbar outside the group.

>     * If you click the kebab icon and open the dropdown of an alert in the drawer, is the dropdown menu expected to display inside of the drawer (if it's not in a group) and/or group (if it's in a group)?
You are correct about the PF3 behavior, but this had actually been raised as a bug. We deferred it because of the difficulty to fix. I do want to do something better for PF4. If there is not enough room to display the menu inside the boundary of the drawer, is it possible to just pop the menu up (above) rather than down? How do we handle this in similar situations, i.e. in a list or table when you open the menu from the last row and there's not enough space to display it?

**Comment by mcoker** [comment]
> If there is not enough room to display the menu inside the boundary of the drawer, is it possible to just pop the menu up (above) rather than down? How do we handle this in similar situations, i.e. in a list or table when you open the menu from the last row and there's not enough space to display it?

We have a variation of the dropdown type components that aligns the menu above the toggle, but whether the component is smart enough to detect the available space and whether the menu should open on the top or bottom of the toggle, I'll need to defer to react devs. @tlabaj do you know?

If we aren't going to have a scrollbar to scroll groups, and the notification item dropdown will open inside the notification drawer and/or group, then this looks like it's ready to build to me. Seems like the dropdown could possibly open outside of the drawer/group, too, if we did something in react that allowed the dropdown menu to exist outside of the notification drawer in the DOM. Maybe something similar to the [appendTo prop for tooltips](http://patternfly-react.surge.sh/patternfly-4/components/tooltip).

**Comment by tlabaj** [comment]
@mcoker The Dropdown component does not detect space and automatically position it above if there is not enough space.  

@jschuler worked on tooltip positioning do you.  Do you think we could do something similar  for Dropdown (see @mcoker comment above)?

**Comment by maryshak1996** [comment]
@mcoker after some discussion with a larger group, there are a few concerns that were raised about the current design that we will likely require some design updates that I wanted to document here, so that you are aware before getting too knee-deep in the development. Here is a break down of the feedback given, and the solutions that I am proposing to resolve the issues can be seen on my updated invision prototype:
- we need a visual indicator associated with the bell icon in masthead to show that the drawer is open
- There is inconsistencies with the badging of 'read' vs 'unread' notifications
- The 'close' icon may not be necessary
- Should we consider relocating the "Mark as read" and "Clear all" actions into the gear? Should the gear instead be a kebab?
- How do we deal with long titles? Truncation? Stacking?
https://redhat.invisionapp.com/share/7FQDO4RP462#/345385129_Notification_Drawer_-_Complex_Example3

**Comment by maryshak1996** [comment]
@mcoker @rachael-phillips I think that all of the feedback from product groups has been considered in these latest mocks above, and we can consider [again] this done from the visual side. sorry for the confusion!

**Comment by maryshak1996** [comment]
a few designers have asked for the sketch file for this, so here she is in all her glory along with the visual spec
[Notification-Drawer.zip](https://github.com/patternfly/patternfly-next/files/2911037/Notification-Drawer.zip)
:

**Comment by mcarrano** [comment]
@maryshak1996 This is looking good. I like your changes. I only had one comment about how we are reflecting the state of the drawer (open vs closed). Did you also pass your Sketch assets along to @gdoyle1 for integration with the master file?

@mcoker please look over this design when you have a chance. I'm wondering how we will approach this from an implementation perspective. I think the drawer itself can be a component. But this also involves integration with an element living in the masthead. Should we also create a demo to show the drawer in context?

**Comment by maryshak1996** [comment]
@mcarrano yes I did! @gdoyle1 has my sketch assets for this

**Comment by mcoker** [comment]
This looks great! Re: implementation, the toggle looks like a button with a badge. Do we want to build something that serves as the badge that can be used elsewhere in the app, or build a notification drawer toggle that has the icon and badge built in?

Also, how is the drawer supposed to be positioned relative to the bottom of the page? Should it extend to the bottom of the viewport? The [PF3 notification drawer](https://rawgit.com/patternfly/patternfly/master-dist/dist/tests/notification-drawer-horizontal-nav.html) nearly extends to the bottom of the page with a 20px gap.

We'll definitely want a demo showing it context. That may need to live in the react workspace, too, since it looks like the positioning of the drawer is going to need to rely on javascript to compute the height.

**Comment by mcarrano** [comment]
@mcoker Addressing your questions...

> 
Do we want to build something that serves as the badge that can be used elsewhere in the app, or build a notification drawer toggle that has the icon and badge built in?

Good question. It's possible that there is a general case for adding a badge to any icon (although I can't think of one offhand), in which case we should make this a component unto itself or a variation of the icon button. @kybaker do you have thoughts on this? The badge applied to an icon is not something we currently have as part of the component set and the existing Badge component does not really work for this.

> 
Also, how is the drawer supposed to be positioned relative to the bottom of the page? Should it extend to the bottom of the viewport?

Yes, the drawer should size to the height of the viewport. I have created an additional issue to create a demo that reflects what the drawer looks like in the content of a page: #1522

**Comment by maryshak1996** [comment]
quick update here: after some stakeholder feedback, I updated the indicator for opened vs unopened drawer in the masthead
Previously the only only masthead indicator was a white outline around the blue 'unread' circle, but the point was raised that the user should see a visual indicator that the drawer is opened even when there is not an unread notification.
What I am proposing now is having the color of the bell itself changing to pf-blue-50. This is a subtle change from the default masthead icon color and we see opened/expanded items turn blue when selected elsewhere (however the selected-blue color is not compliant over the masthead color, hence using the lightest shade of blue). Here's what it would look like:
![Screen Shot 2019-03-11 at 10 38 41 AM](https://user-images.githubusercontent.com/19824574/54132669-07474e80-43eb-11e9-861a-2adace665566.png)
My invision above is updated to reflect this

**Comment by mcarrano** [comment]
This is pretty subtle @maryshak1996 . Not sure that I would have noticed a difference if you did not point it out. What do others think?

**Comment by matthewcarleton** [comment]
I agree with @mcarrano 
Is the intention for the circle to be present when the drawer is open? I could see removing it and changing the bell icon to blue. 
If I've opened the drawer and "seen" all of the new notifications but haven't explicitly marked them as read does that circle remain when I close the drawer? 
I could see an argument for removing it and then adding it again if another notification is present. Alternatively the circle could be removed when the drawer is open and replaced when it is closed again, which would allow more flexibility on the design of the bell icon.

**Comment by maryshak1996** [comment]
@mcarrano @matthewcarleton 
I do agree that it's very sublte, and I think that this something that we may want to consider addressing more globally in reference to the 'selected vs unselected' behavior of all utility icons (so the cog as well, launcher, or other icons that various products might use up there). What do you think?
With that said, I think that we should have the 'selected/unselected' visual indicator be seperate from the blue dot, since that is not something that other icons would have nor something that would appear when there are 0 unread notifications -- thinking about the 'selected/unselected' visual treatment should probably be considered externally from that blue dot for these reasons in order to have a standard that is consistent across utility icons

**Comment by maryshak1996** [comment]
I'll keep tweaking how we represent 'selected/unselected' with the bell for now and reach out to a few other visual folks too :)

**Comment by rachael-phillips** [comment]
Hi @maryshak1996 how is this issue going?

**Comment by maryshak1996** [comment]
@rachael-phillips hey sorry about the hold up on this! I have a couple of options for this last piece (opened vs unopened utility item), that I'm just working to refine :) should have the final design recommendation here shortly!

**Comment by mcarrano** [comment]
@rachael-phillips I also have this in my queue to work on an interaction spec. But once Mary is done, I think there will be enough to hand over to CSS if/when there's a resource to take it.

**Comment by maryshak1996** [comment]
@rachael-phillips here's what we'd like to do!
![image](https://user-images.githubusercontent.com/19824574/54935990-e0633f00-4ef7-11e9-82cf-941eb7207b1b.png)

**Comment by rachael-phillips** [comment]
That sounds great. Thanks for the update!

**Comment by karelhala** [comment]
This component looks good! I really like the guidelines you are working on here, however let's not tangle together icon, drawer and notifications together.

Let me propose an idea and let's think about it to see if it's something reasonable or not.
#### Notification icon/text
I really like the idea of icon that indicates some changes has been done or happend since user last went to specific part of application, so I would really like to see a component that allows me to add icon ro text and I'd see this icon with blue circle at top to indicate that user should click on it and see what is there.

#### Drawer
Please do not think about drawer as notification drawer specific, this is really good design guideline, but not good component. We should have a component that is just static drawer opened on any place with static width and some other properties [1]. What the consumer is going to add to this component is really not our business, we can guide them and tell them which components are good, but they can choose whatever they want in there.

#### Noification drawer
With these things said I think that the best approach to implement notification drawer would be to implement it using these components
* Notification icon - new component
* Drawer on right side - new component
  * Accordion [2] - different header, but same component
    * Notification group [3]

Why these changes you might ask? What if someone doesn't realy likes bell icon, but would rather use inbox icon? What if they want to position the drawer on left side? What if someone would like to use sortable table with collapsible items as notification groups? All these questions might come in future we'd have to write specific components, or consumers will have to write specific component based on what they really want. Let's give developers freedom of choice and provide them with guidelines, but allow them to break these guidelines.

[1] https://material-ui.com/demos/drawers/ 
[2] https://pf4.patternfly.org/components/Accordion/examples/
[3] https://pf4.patternfly.org/components/AlertGroup/examples/

**Comment by mcarrano** [comment]
@karelhala I like this proposal. While I'm not in favor of having many variations of the Notification Drawer pattern, I agree that, in general, flexible/modular design is a good thing. I can see more generalized components like a notification icon and a sliding drawer as applicable to a variety of use cases, and therefore it makes sense to build in that way. 

@mcoker, @matthewcarleton  and @tlabaj I'd be interested to know what you think? Ultimately I see this as a development decision, but I can easily factor the design into more modular pieces if you think that makes sense. The Notification Drawer would then be more like a demo that puts these pieces together.

**Comment by matthewcarleton** [comment]
I see a lot of value in breaking this out since we will have use for these things on their own. My biggest question is can we implement this in a demo if we have specific needs on the accessibility side? @jgiardino based on the work you've done so far with alerts do you have any thoughts on this?

**Comment by jgiardino** [comment]
I'm also in favor of @karelhala's proposal.

@matthewcarleton The main a11y thing with dynamic alerts is handling the live region. But the notification drawer shouldn't need to include live regions. The a11y patterns that I think apply here are the relationship of the icon in the masthead with the drawer, and what role the drawer has. It isn't clear to me if the notification drawer displays besides other contents or on top of other contents. If it's beside contents, we would probably use `role="aside"`. But if it's on top of contents, the approach I've been taking is to use a similar pattern that we use for modals or popovers, which uses `role="dialog"` and traps focus.

**Comment by matthewcarleton** [comment]
thanks @jgiardino That makes a lot of sense.
I think we'd just have to figure out how this differs from the current "alert group". I can see some confusion there if we have a component for the notifications inside the draw and then another one for the alerts outside of it when they are grouped. Probably just needs some brainstorming :)

**Comment by mcoker** [comment]
+1 to @karelhala's idea of separating into components, sounds like a smart approach.

**Comment by mcarrano** [comment]
Based on the ideas included [here](https://docs.google.com/document/d/1DAJL7RnG759cg6ey0BXxGlQShGDcORk2_Nx6h8gWV4w/edit?usp=sharing) and some follow-up discussion, some other items should be considered. This include:

* Whether we need a way to restore an alert that was cleared accidentally.
* How to display a neutral alert state. Issue #1749 
* Mobile considerations?
* Filtering drawer contents

**Comment by LHinson** [comment]
Thanks for the suggestions on how to break this component down @karelhala 
@mcoker I've created, linked and assigned you those issues.

**Comment by mcoker** [comment]
Latest designs from @maryshak1996 https://redhat.invisionapp.com/share/7FQDO4RP462#/345357488_Visual_Spec

**Comment by mcarrano** [comment]
I have added the demo tag to this issue and will close #1522 as a duplicate now that we have broken this into smaller components. I agree with the decomposition but we will still want to have a demo that shows these pieces working together.

**Comment by maryshak1996** [comment]
I've updated the invision and visual spec based on some collaboration with @mceledonia

**Comment by maryshak1996** [comment]
https://sketch.cloud/s/y50lz

**Comment by maryshak1996** [comment]
With the switch to marvel, the most up to date stuff can be found here: https://marvelapp.com/project/4090736/

**Comment by mcarrano** [comment]
Draft of usage document can be found here: https://docs.google.com/document/d/1kccmdwXgnkLZdH5OY_sr6C91Gsm0gCtF6zp9OBwC-dE/edit?usp=sharing

**Comment by karelhala** [comment]
I really want to stress out that we can't limit ourselves to drawer itself. I should be able to place notifcation accordion wherver I want - dialog, full page w/e. And drawer itself can contain whatever we want. I guess that everybody already know how drawer will look like side component that is over other component. The drawer should be flexible enough to allow app devs to set its width (either percentage of page or fixed doesn't matter all the way to full page view). And anything can be part of drawer it doesn't really have to be notifications. We should start with drawer itself and once it is done we will understand much more other parts of this notification drawer demo.

**Comment by mcarrano** [comment]
@karelhala Thanks for your detailed comments on the design/usage doc that I posted above. @mcoker @dlabaj please take note as they mostly address how the drawer component is implemented.

**Comment by mcoker** [comment]
@karelhala thanks for the feedback. Have you seen the [drawer component](https://pf4.patternfly.org/components/Drawer/examples/) yet? Also here is a [demo of the drawer in the page component](https://pf4.patternfly.org/demos/Drawer/examples/).

The notification list is something that will be placed inside of the existing drawer component. Do you see any problems with that?

**Comment by karelhala** [comment]
@mcoker was not aware of the drawer component, nice! Yes, that is absolutely correct to place notification list in the drawer, just bear in mind that it can be placed anywhere else.

**Comment by mcoker** [comment]
closing in favor of https://github.com/patternfly/patternfly/issues/3186
